app/main.py

```diff
@@ -1,6 +1,20 @@
 from typing import Callable
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached_results = {}
+
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> int:
```

Comment:
The return type could be `Any` because the output of the wrapped function is not known in advance to the decorator.